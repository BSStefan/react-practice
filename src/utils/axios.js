// @flow
import axios from 'axios';

export const setTmdbKey = (key: string): void => {
  axios.defaults.params = {};
  axios.defaults.params['api_key'] = key;
};

export const setBaseUrl = (url: string): void => {
  axios.defaults.baseURL = url;
}