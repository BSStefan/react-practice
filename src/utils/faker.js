// @flow
import type { Movie } from '../types/Movie';

export const fakeMovie = ({ id, title, original_language, genre_ids, overview, poster, release_date }: Movie = {}): Movie => ({
  id: id || 123,
  title: title || 'Movie title',
  original_language: original_language || 'en',
  genre_ids: genre_ids || ['123'],
  overview: overview || 'Test test',
  poster: poster || 'http://test.com',
  release_date: release_date || '12/12/2019',
});
