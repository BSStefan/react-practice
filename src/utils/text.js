// @flow

export const shortText = (text: string, maxLenght: number): string => text.length > maxLenght ? `${text.slice(0, maxLenght - 5)} ...` : text;