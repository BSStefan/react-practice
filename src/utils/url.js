export const TMDB_URL = 'https://api.themoviedb.org/3';
export const TMDB_POSTER_URL = 'https://image.tmdb.org/t/p/w185_and_h278_bestv2'