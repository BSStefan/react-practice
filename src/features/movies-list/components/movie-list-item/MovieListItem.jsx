// @flow
import * as React from 'react';
import styled from 'styled-components';
import { Card, Button } from 'react-bootstrap';
import { TMDB_POSTER_URL } from '../../../../utils/url';
import { shortText } from '../../../../utils/text';
import type { Movie } from '../../../../types/Movie';

type Props = {
  movie: Movie
}

const MovieListItem = (props: Props) => {
  const { movie } = props;
  return (
    <StyledCard id={movie.id}>
      {movie.poster && <Card.Img variant="top" src={`${TMDB_POSTER_URL}${movie.poster}`} />}
      <Card.Body>
        <Card.Title>{movie.title}</Card.Title>
        <Card.Text>{shortText(movie.overview, 100)}</Card.Text>
        <Button variant="primary">View Movie</Button>
      </Card.Body>
    </StyledCard>
  )
};

const StyledCard = styled(Card)`
    margin-bottom: 15px;
  `

export default MovieListItem;