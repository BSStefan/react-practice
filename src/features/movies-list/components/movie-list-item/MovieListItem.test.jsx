import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import MovieListItem from './MovieListItem';
import { fakeMovie } from '../../../../utils/faker';


const defaultProps = {
  movie: fakeMovie()
}

it('Should match snapshoot with poster', () => {
  const tree = renderer.create(<MovieListItem {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
})

it('Should match snapshoot without poster', () => {
  const defaultPropsWithoutPoster = { movie: { ...defaultProps.movie, poster: undefined } };
  const tree = renderer.create(<MovieListItem {...defaultPropsWithoutPoster} />).toJSON();

  expect(tree).toMatchSnapshot();
})