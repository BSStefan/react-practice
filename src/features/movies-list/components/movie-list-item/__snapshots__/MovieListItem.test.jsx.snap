// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshoot with poster 1`] = `
<div
  className="sc-bdVaJa hniPbm card"
  id={123}
>
  <img
    className="card-img-top"
    src="https://image.tmdb.org/t/p/w185_and_h278_bestv2http://test.com"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Movie title
    </div>
    <p
      className="card-text"
    >
      Test test
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      View Movie
    </button>
  </div>
</div>
`;

exports[`Should match snapshoot without poster 1`] = `
<div
  className="sc-bdVaJa hniPbm card"
  id={123}
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Movie title
    </div>
    <p
      className="card-text"
    >
      Test test
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      View Movie
    </button>
  </div>
</div>
`;
